server/pbench/bin/index-pbench

```diff
@@ -393,19 +680,113 @@ def mk_tool_data(mdlog, md5sum, tb, tstamp):
             for host in hosts:
                 tools = get_tools(iteration, sample, host, mdlog, tb)
                 for tool in tools:
-                    if tool == 'pidstat':
-                        # FIXME - Skipping indexing of pidstat data for now
-                        continue
-                    data_files = get_csv_files(
-                        iteration, sample, host, tool, mdlog, tb)
-                    for dfile in data_files:
-                        yield ToolData(
-                            tstamp, md5sum, experiment, iteration, sample,
-                            host, tool, toolsgroup, os.path.basename(dfile),
-                            dfile, tb)
+                    yield ToolData(
+                        tstamp, md5sum, experiment, iteration, sample,
+                        host, tool, toolsgroup, mdlog, tb)
+    return
+
+def mk_tool_data_actions(ptb, options, INDEX_PREFIX):
+    index_name_template = "%s.tool-%%s" % INDEX_PREFIX
+    for td in mk_tool_data(ptb.mdconf, ptb.md5sum, ptb.tb, ptb.start_run):
+        # Each ToolData object, td, that is returned here represents how
+        # data collected for that tool across all hosts is to be returned.
+        # The make_source method returns a generator that will emit each
+        # source document for the appropriate unit of tool data.  Each has
+        # the option of constructing that data as best fits its tool data.
+        # The tool data for each tool is kept in its own index to allow
+        # for different curation policies for each tool.
+        index_name_template_for_tool = (index_name_template % td.toolname) + '.'
+        asource = td.make_source()
+        if not asource:
+            continue
+        for source in asource:
+            action = {
+                "_op_type": "create",
+                "_index":   index_name_template_for_tool + source['@timestamp'].split('T', 1)[0],
+                "_type":    "pbench-tool-data",
+                # FIXME: we should generate our own JSON doc ID
```

Comment:
This would be very helpful to have ahead of time.  I think the hash should include all the metadata, timestamp, field names and values.